packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -257,22 +257,22 @@ class _HeroFlight {
       animation: _proxyAnimation,
       child: manifest.toHero.widget,
       builder: (BuildContext context, Widget child) {
-        final RenderBox toHeroBox = manifest.toHero.context?.findRenderObject();
-        if (_aborted || toHeroBox == null || !toHeroBox.attached) {
-          // The toHero no longer exists or it's no longer the flight's destination.
+        final RenderBox finalHeroBox = manifest.toHero.context?.findRenderObject();
+        if (_aborted || finalHeroBox == null || !finalHeroBox.attached) {
+          // The finalHero no longer exists or it's no longer the flight's destination.
           // Continue flying while fading out.
           if (_heroOpacity.isCompleted) {
             _heroOpacity = new Tween<double>(begin: 1.0, end: 0.0)
               .chain(new CurveTween(curve: new Interval(_proxyAnimation.value, 1.0)))
               .animate(_proxyAnimation);
           }
-        } else if (toHeroBox.hasSize) {
-          // The toHero has been laid out. If it's no longer where the hero animation is
-          // supposed to end up (heroRect.end) then recreate the heroRect tween.
-          final RenderBox routeBox = manifest.toRoute.subtreeContext?.findRenderObject();
-          final Offset heroOriginEnd = toHeroBox.localToGlobal(Offset.zero, ancestor: routeBox);
-          if (heroOriginEnd != heroRect.end.topLeft) {
-            final Rect heroRectEnd = heroOriginEnd & heroRect.end.size;
+        } else if (finalHeroBox.hasSize) {
+          // The finalHero has been laid out. If it's no longer where the hero animation is
+          // supposed to end up then recreate the heroRect tween.
+          final RenderBox finalRouteBox = manifest.toRoute.subtreeContext?.findRenderObject();
+          final Offset finalHeroOrigin = finalHeroBox.localToGlobal(Offset.zero, ancestor: finalRouteBox);
+          if (finalHeroOrigin != heroRect.end.topLeft) {
+            final Rect heroRectEnd = finalHeroOrigin & heroRect.end.size;
```

Comment:
I don't feel super strongly about this but I think we should probably revert these back to using the word "to" instead of "final" so that they match "toHero".